job/replay_syncer.go

```diff
@@ -0,0 +1,181 @@
+package job
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/odpf/optimus/core/logger"
+	"github.com/odpf/optimus/core/tree"
+	"github.com/odpf/optimus/models"
+	"github.com/odpf/optimus/store"
+	"github.com/pkg/errors"
+)
+
+var (
+	ReplayStatusToSynced = []string{models.ReplayStatusReplayed, models.ReplayStatusInProgress, models.ReplayStatusAccepted}
+	ReplayMessageSuccess = "all instances for this replay are successfully run"
+	ReplayMessageFailed  = "instance run failure found"
+)
+
+type ReplaySyncer interface {
+	Sync(context.Context, time.Duration) error
+}
+
+type Syncer struct {
+	replaySpecFactory         ReplaySpecRepoFactory
+	scheduler                 models.SchedulerUnit
+	dependencyResolver        DependencyResolver
+	projectJobSpecRepoFactory ProjectJobSpecRepoFactory
+	assetCompiler             AssetCompiler
+	Now                       func() time.Time
+	projects                  []models.ProjectSpec
+}
+
+func NewReplaySyncer(replaySpecFactory ReplaySpecRepoFactory, scheduler models.SchedulerUnit, dependencyResolver DependencyResolver, projectJobSpecRepoFactory ProjectJobSpecRepoFactory, assetCompiler AssetCompiler, projects []models.ProjectSpec) *Syncer {
```

Comment:
We can format these lines to have a soft limit to 80 and a hard limit to 120 chars

Reply:
i see you have refactored this, thanks!